chcemvediet/apps/inforequests/models/inforequest.py

```diff
@@ -544,7 +544,9 @@ def get_absolute_url(self, anchor=u''):
     def _send_notification(self, template, anchor, dictionary):
         dictionary.update({
                 u'inforequest': self,
-                u'url': complete_url(self.get_absolute_url(anchor)),
+                u'url': u'{}?next={}'.format(complete_url(reverse(u'account_login')),
```

Comment:
To nemozeme takto naivne zlepit stringy, ved hodnoty do query treba escapovat, inac to bude blbnut.

Vid napriklad `chcemvediet/apps/inforequests/sitemaps.py` ako sa escapuju url paramere pomocou `urlencode`

Reply:
@martinmacko Nerozumiem. Ja spajam 2 existujuce url: `complete_url(reverse(u'account_login'))` a  `self.get_absolute_url(anchor)`.

String vrateni z `reverse()` *(get_absolute_url) uz je **urlquoted**. Napriklad:
```
>>> moja_infoziadost.get_absolute_url('#a2')
'/inforequests/moja-infoziadost-1/#a2''
```
Opatovne pouzit encoding na vysledok z `get_absolute_url` je neziaduce.

```
>>> urlencode({'next':'/inforequests/moja-infoziadost-1/#a2'})
'next=%2Finforequests%2Fmoja-infoziadost-1%2F%23a2'
```

A ano, krajsi zapis by bol bez `.format`
```
url = complete_url(reverse(u'account_login')) + u'?next=' + self.get_absolute_url(anchor)
```

Reply:
Pozri **Note** v casti [reverse](https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#reverse).

Reply:
@viliambalaz Ano, url vratena z `self.get_absolute_url(anchor)` ma svoje parametre encodovane. Ale potom tu url pouzivame ako parameter do inej url.

Napriklad ten `#a2`. Volanie `self.get_absolute_url(anchor)` vrati string `/inforequests/moja-infoziadost-1/#a2`. Ak ten vlozis do povodnej URL bez escapovania, dostanes `https://www.chcemvediet.sk/sk/accounts/login/?next=/inforequests/moja-infoziadost-1/#a2`.

Ku ktorej URL teraz to `#a2` patri? Ku tej co je v parametre `next` alebo ku ten vonkajsej URL? Bez escapovania to browser pochopi, ze to patri k tej vonkajsej URL a na server to ani neposle.

Ale iny priklad, predstav si, ze `self.get_absolute_url(anchor)` by vratilo URL `/path?foo=aaa&bar=bbb`. Co keby si dal so `next` bez escapovania, dostal by si URL: `https://www.chcemvediet.sk/sk/accounts/login/?next=/path?foo=aaa&bar=bbb`. Skus si taku URL dat rozparsovat, ake parametre pre vonkajsiu URL z toho budu? Hodnota `next` nebude `/path?foo=aaa&bar=bbb` ako by si ocakaval, ale hodnota `next` bude `/path?foo=aaa`, pretoze `&` znamena, ze to co je za nim je dalsi parameter.